ocaml/database/db_upgrade.ml

```diff
@@ -40,6 +39,11 @@ let generic_database_upgrade db =
          Table.add g objref row tbl in
        let tbl = Table.fold add_fields_to_row tbl Table.empty in
        let g = Manifest.generation (Database.manifest db) in
-       ((Database.update ++ (TableSet.update g tblname Table.empty)) (fun _ -> tbl)) db
+       let perform_update =
+         (fun _ -> tbl)
```

Comment:
if I read this right then this effectively ignores the `db` parameter below?
Could this be simplified to ```
tbl
|> TableSet.update g tblname Table.empty
|> Database.update
```?

Reply:
I have left it because I had fear of unexpected side effects, like some globals changing behind the back. I think it can be simplified but I'd like to double check the code and maybe ping @jonludlam